### PR TITLE
Add 07-mars-viewer homework

### PR DESCRIPTION
**Pull request by kgalushchak**

### CODE REVIEW

File: 07-mars-viewer/src/features/photos/Photos.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+import { useAppSelector, useAppDispatch } from '../../app/hooks';
+import { getPhotos,  getSol, setSol, selectPhotos } from './photosSlice';
+import styles from './Photos.module.css';
+import {Gallery} from './Gallery';
+
+export function Photos() {
+  const photos = useAppSelector(selectPhotos);
+  const sol = useAppSelector(getSol);
```
Comment (ivankoleda) [inline 07-mars-viewer/src/features/photos/Photos.tsx]: selector function names should start from `select`

File: 07-mars-viewer/src/features/photos/photosSlice.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+export interface GalleryProps {
+  photos: PhotoProps[];
+}
+
+export interface PhotosState {
+  sol: string;
+  photos: PhotoProps[];
+  favourites: PhotoProps[];
```
Comment (ivankoleda) [inline 07-mars-viewer/src/features/photos/photosSlice.ts:18]: based on concept of single source of truth it's better to store only ids of favourite photos and not duplicate the whole photo, but for that photos state also need to be rewritten to store photos from all requests

File: 07-mars-viewer/src/features/photos/photosSlice.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    const response = await getMarsPhotos(sol);
+    return response;
+  }
+);
+
+export const photosSlice = createSlice({
+  name: 'photos',
+  initialState,
+  reducers: {
+    manageFavourites: (state, action: PayloadAction<PhotoProps>) => {
```
Comment (ivankoleda) [inline 07-mars-viewer/src/features/photos/photosSlice.ts]: better call `toggleFavourites`

File: 07-mars-viewer/src/App.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import React, {useState} from 'react';
+import { Photos } from './features/photos/Photos';
+import { Gallery } from './features/photos/Gallery';
+import { useAppSelector } from './app/hooks';
+import { selectFavourites } from './features/photos/photosSlice';
+import './App.css';
+
+function App() {
+  const TAB_PHOTOS = 'Photos';
+  const TAB_FAVOURITES = 'Favourites';
```
Comment (ivankoleda) [inline 07-mars-viewer/src/App.tsx]: better use `enum` for such cases

**Review by ivankoleda — COMMENTED** [review]
<h1>Review</h1>

<h2>Good parts</h2>

* Everything works well and according to task description
* Types written well
* Usage of env variable
* UI 

<h2>What can be improved</h2>

* better not to copy and create duplicates in favourite state
* no example of usage `createSelector`

File: 07-mars-viewer/src/features/photos/Photos.tsx (hunk 1 above)
Comment (kgalushchak) [reply 07-mars-viewer/src/features/photos/Photos.tsx]: Renamed

File: 07-mars-viewer/src/features/photos/photosSlice.ts (hunk 3 above)
Comment (kgalushchak) [reply 07-mars-viewer/src/features/photos/photosSlice.ts]: Renamed

File: 07-mars-viewer/src/App.tsx (hunk 4 above)
Comment (kgalushchak) [reply 07-mars-viewer/src/App.tsx]: Created enum

File: 07-mars-viewer/src/features/photos/photosSlice.ts (hunk 2 above)
Comment (kgalushchak) [reply 07-mars-viewer/src/features/photos/photosSlice.ts:18]: Will try to fix in Mobx version of the app

**Pull request merged**